src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1503,7 +1503,7 @@ public virtual BadRequestObjectResult BadRequest(ModelStateDictionary modelState
         /// </summary>
         /// <returns>The created <see cref="BadRequestObjectResult"/> for the response.</returns>
         [NonAction]
-        public virtual BadRequestObjectResult ValidationProblem(ValidationProblemDescription descriptor)
+        public virtual BadRequestObjectResult ValidationProblem(ValidationProblemDetails descriptor)
```

Comment:
Should we return `ActionResult` here? You can override this but you can't change the result type.

Reply:
Sure.